packages/medusa/src/api/admin/fulfillments/validators.ts

```diff
@@ -32,13 +32,13 @@ export const AdminCreateFulfillment = z.object({
   labels: z.array(AdminCreateFulfillmentLabel),
   order: z.object({}),
   order_id: z.string(),
-  shipping_option_id: z.string().optional(),
-  data: z.record(z.unknown()).optional().nullable(),
-  packed_at: z.coerce.date().optional().nullable(),
-  shipped_at: z.coerce.date().optional().nullable(),
-  delivered_at: z.coerce.date().optional().nullable(),
-  canceled_at: z.coerce.date().optional().nullable(),
-  metadata: z.record(z.unknown()).optional().nullable(),
+  shipping_option_id: z.string().nullish(),
+  data: z.record(z.unknown()).nullable(),
+  packed_at: z.coerce.date().nullish(),
+  shipped_at: z.coerce.date().nullish(),
+  delivered_at: z.coerce.date().nullish(),
+  canceled_at: z.coerce.date().nullish(),
+  metadata: z.record(z.unknown()).nullable(),
```

Comment:
why are soem of these not `nullish`? `e.g. `metadata` is not required

Reply:
Good catch, replace all screw up. Some modules are setup for metadata to not be nullable. Shall we make them all `metadata: z.record(z.unknown()).nullish()`?

Reply:
I would say so yeah